### PR TITLE
docs: Optimized readme.md code format

### PR DESCRIPTION
**Pull request by AwesomeDevin**
## Summary
Optimized readme.md code format

## Check List

- [x ] `yarn run prettier` for formatting code and docs

### CODE REVIEW

**Comment by dai-shi** [comment]
The partial code is _intentional_ to reduce the lines of readme. Does it seem too confusing?

btw, I didn't know prettier didn't work for readme.md.

File: readme.md [hunk 1]
```diff
@@ -96,17 +100,16 @@ const sum = atom((get) => atoms.map(get).reduce((acc, count) => acc + count))
...
+const urlAtom = atom('https://json.host.com')
+const fetchUrlAtom = atom(async (get) => {
+  const response = await fetch(get(urlAtom))
+  return await response.json()
+})
 
 function Status() {
   // Re-renders the component after urlAtom changed and the async function above concludes
   const [json] = useAtom(fetchUrlAtom)
+}
```
Comment (dai-shi) [inline readme.md:112]: This one looks a bit strange.

**Comment by AwesomeDevin** [comment]
> The partial code is _intentional_ to reduce the lines of readme. Does it seem too confusing?
> 
> btw, I didn't know prettier didn't work for readme.md.

The partial code looks a little uncomfortable， Maybe it's just my personal habit ～ 😂

File: readme.md (hunk 1 above)
Comment (dai-shi) [reply readme.md:112]: That seems over-complicated. We want to keep readme very compact, and move detailed things into `./docs/`.

**Comment by dai-shi** [comment]
> Maybe it's just my personal habit

I heard similar comments in [zustand](https://github.com/pmndrs/zustand), so you are not alone, though.

File: readme.md [hunk 2]
```diff
@@ -53,7 +53,10 @@ function Counter() {
   return (
     <h1>
       {count}
-      <button onClick={() => setCount(c => c + 1)}>one up</button>
+      <button onClick={() => setCount((c) => c + 1)}>one up</button>
+    </h1>
+  )
+}
```
Comment (dai-shi) [inline readme.md:59]:
```suggestion
      <button onClick={() => setCount((c) => c + 1)}>one up</button>
      ...
```

How about adding `...` to make it explicit that it's partial code?

Comment (AwesomeDevin) [reply readme.md:59]: good idea~  ， It works

File: readme.md (hunk 1 above)
Comment (AwesomeDevin) [reply readme.md:112]: Got it  , thanks .

File: readme.md (hunk 2 above)
Comment (AwesomeDevin) [reply readme.md:59]: Shall I modify it and commit merge request  again?

Comment (dai-shi) [reply readme.md:59]: Yeah, please modify entirely with the said policy.

Comment (AwesomeDevin) [reply readme.md:59]: Got it~！

File: readme.md [hunk 3]
```diff
@@ -126,18 +127,22 @@ function Counter() {
...
 
 ### Write only atoms
 
 Just do not define a read function.
 
 ```jsx
-const multiplyCountAtom = atom(null, (get, set, by) => set(countAtom, get(countAtom) * by))
+const multiplyCountAtom = atom(null, (get, set, by) =>
+  set(countAtom, get(countAtom) * by)
+)
```
Comment (dai-shi) [inline readme.md:140]: Can we change the name to make it one line? For example, `multiplyCountAtom` 👉 `multiplyAtom`

File: readme.md [hunk 4]
```diff
@@ -155,7 +160,9 @@ const fetchCountAtom = atom(
 
 function Controls() {
   const [count, compute] = useAtom(fetchCountAtom)
-  return <button onClick={() => compute("http://count.host.com")}>compute</button>
+  return (
+    <button onClick={() => compute('http://count.host.com')}>compute</button>
+    ...
```
Comment (dai-shi) [inline readme.md:165]:
```suggestion
  return <button onClick={() => compute('http://count.host.com')}>compute</button>
}
```
is shorter if prettier allows it.

**Review by dai-shi — COMMENTED** [review]
Getting good 👍 ! Added two comments.

File: readme.md (hunk 3 above)
Comment (AwesomeDevin) [reply readme.md:140]: It's fine ，but if `run prettier`，it will be wrapped. 😂😂😂

Comment (dai-shi) [reply readme.md:140]: even if you used the shorter name?

Comment (AwesomeDevin) [reply readme.md:140]: Yes, I tried，Unless use `multi`

```js
const multi = atom(null, (get, set, by) => set(countAtom, get(countAtom) * by))

function Controls() {
  const [, multi] = useAtom(multi)
  return <button onClick={() => multi(3)}>triple</button>
}
```

Comment (dai-shi) [reply readme.md:140]: Oookay, let's leave it then.

File: readme.md (hunk 4 above)
Comment (dai-shi) [reply readme.md]: oh, no. this one fails too.
so, i guess 2fe941c is good. sorry to bother.

File: readme.md (hunk 3 above)
Comment (AwesomeDevin) [reply readme.md:140]: 👌

**Comment by dai-shi** [comment]
https://github.com/pmndrs/jotai/pull/1681#discussion_r1068974930
Can you revert to https://github.com/pmndrs/jotai/commit/2fe941c3e2f1dc9cdbd269d07ee3f56350ea529e ?

**Comment by AwesomeDevin** [comment]
> [#1681 (comment)](https://github.com/pmndrs/jotai/pull/1681#discussion_r1068974930) Can you revert to [2fe941c](https://github.com/pmndrs/jotai/commit/2fe941c3e2f1dc9cdbd269d07ee3f56350ea529e) ?


Got it!

**Review by dai-shi — APPROVED** [review]
Thanks!!!

**Comment by dai-shi** [comment]
Unfortunately, we need to revert this too:
https://github.com/pmndrs/jotai/compare/2fe941c...43a6846

**Comment by AwesomeDevin** [comment]
> 



> Unfortunately, we need to revert this too: [2fe941c...43a6846](https://github.com/pmndrs/jotai/compare/2fe941c...43a6846)

😂  I resubmitted a copy of the code，is it ok?

**Comment by dai-shi** [comment]
Should be good. Thanks!

**Comment by AwesomeDevin** [comment]
> Should be good. Thanks!

Thank you for your guidance ~

**Pull request merged**